modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/DateFormat.java

```diff
@@ -97,6 +97,7 @@ Function<String, ZonedDateTime> getFunction(String format, ZoneId zoneId, Locale
                 // fill the rest of the date up with the parsed date
                 if (accessor.isSupported(ChronoField.YEAR) == false
                     && accessor.isSupported(ChronoField.YEAR_OF_ERA) == false
+                    && accessor.isSupported(WeekFields.ISO.weekBasedYear()) == false
```

Comment:
These additions, and similar, are because `ISO.weekBasedYear` is a different thing to `of(locale).weekBasedYear`, even though they probably represent the same thing